### ISSUE TITLE
Non-default Click actions not working

### ISSUE THREAD

**Issue by kewlgrapes** [micheleg/dash-to-dock#414]
I am using dash-to-dock v45 and gnome-shell 3.14.4 on RHEL 7.3. I set the Click action in the Behavior settings to "Launch new instance", but nothing happens when I click on an icon to start a second instance of an application. I also tried "minimize" and "Cycle through windows", but neither one appears to do anything.

Not a big deal because I can just CTRL+click to launch additional instances of an application, but it would be nice to save the extra keystroke.

Thank you for creating this wonderful extension!

**Comment by micheleg** [comment]
Hi,

sorry for the late reply. the latest available version for GNOME Shell 3.14 is version 51:

https://micheleg.github.io/dash-to-dock/releases.html

Would you mind checking if the problem persists?

**Comment by kewlgrapes** [comment]
I updated to version 51 and am still seeing the same behavior.

**Comment by phw** [comment]
I also cannot get any of the special actions working. Left mouse click is always "raise", middle click always "new window", no matter which action I choose in the extensions settings.

Nothing in the logs, either.

Using Dash to Dock 56 on Gnome 3.22.2 on Arch Linux.

**Comment by micheleg** [comment]
This is strange, I cannot reproduce this issue. Anything "strange" in your configuration? Maybe some conflict with other extensions?

**Comment by kewlgrapes** [comment]
I tried disabling all other extensions, resetting dash-to-dock's configuration with `dconf reset`, and did a `gnome-shell restart`, but was still able to reproduce the problem.

Looking glass is reporting no errors from dash-to-dock. I did a quick look through some of the logfiles and nothing stood out to me, but I'll admit I'm not really sure what messages to expect/not expect in those files...

By the way, I am running in a dual monitor setup. This is my configuration:

    [nick@thinkpad ~]$ dconf dump /org/gnome/shell/extensions/dash-to-dock/
    [/]
    show-apps-at-top=false
    customize-click=false
    extend-height=false
    dock-fixed=true
    insensitive-message-tray=false
    click-action='launch'
    scroll-switch-workspace=false
    show-show-apps-button=true
    scroll-switch-workspace-whole=false
    show-favorites=true
    apply-custom-theme=false
    dock-position='RIGHT'
    custom-theme-shrink=true
    background-opacity=0.23000000000000001
    height-fraction=0.93999999999999995
    dash-max-icon-size=44
    opaque-background=true
    opaque-background-always=true
    show-running=false

**Comment by micheleg** [comment]
Are you still experiencing this issue?

**Comment by phw** [comment]
Just an update that I still have this in version 60. No way to change the behavior of any click action (neither the default single click nor the advanced click options with shift and middle button). No error message in the journal or lg

Changing the scroll action works as expected. 

Nothing special with my setup AFAIK:

- Gnome 3.24 on up-to-date Arch Linux
- German locale
- Tried disabling all extension and restarting shell (with rs) without any change

**Comment by franglais125** [comment]
@phw can you include a dconf dump? I'll try to test it tomorrow on 3.24.

`dconf dump /org/gnome/shell/extensions/dash-to-dock/`

**Comment by phw** [comment]
Here we go:

```
[/]
animation-time=0.10000000000000001
preferred-monitor=0
custom-theme-running-dots=false
hide-delay=0.40000000000000002
height-fraction=0.94999999999999996
multi-monitor=false
show-running=true
show-delay=0.10000000000000001
customize-click=false
show-show-apps-button=true
show-windows-preview=true
custom-background-color=false
middle-click-action='launch'
shortcut=['<Super>q']
show-favorites=true
scroll-action='cycle-windows'
intellihide-mode='FOCUS_APPLICATION_WINDOWS'
opaque-background=false
scroll-switch-workspace-whole=true
shift-click-action='minimize'
force-straight-corner=true
background-opacity=0.80000000000000004
background-color='#204a87'
show-apps-at-top=false
isolate-workspaces=false
shift-middle-click-action='previews'
scroll-switch-workspace-one-at-a-time=false
dock-fixed=false
autohide-in-fullscreen=false
custom-theme-customize-running-dots=false
click-action='cycle-windows'
extend-height=false
custom-theme-running-dots-color='#ffffff'
isolate-monitors=false
apply-custom-theme=false
animate-show-apps=true
custom-theme-running-dots-border-color='#ffffff'
custom-theme-shrink=true
dock-position='LEFT'
icon-size-fixed=false
dash-max-icon-size=41
intellihide-perapp=false
```

**Comment by franglais125** [comment]
From what I see (haven't tested) you have the `customize-click` option set to false (and so does @kewlgrapes). Can you try setting it to true with dconf?

**Comment by phw** [comment]
Indeed, activating `customize-click` does the trick. How is this option supposed to be set? I don't see anything obvious in the settings dialog.

Might be worth mentioning that I use dash to dock for a long time now (since Gnome 3.14), so maybe some update did not go right

**Comment by franglais125** [comment]
I'm actually not sure about that setting. I started using dash-to-dock on Debian Jessie (3.14), and I upgraded from that just like you (straight to 3.22 in my case). I don't know if its value was changed at some point.

Anyhow, sorry I didn't catch this earlier, but happy we sorted it out!

**Comment by kewlgrapes** [comment]
Enabling the `customize-click` option resolved the issue for me as well. Thanks for the tip! My CTRL key is eternally grateful.

**Comment by micheleg** [comment]
It looks like we left some old settings key  around. We used to have a main switch in the settings to disable click customization all together. So this key should be removed now.

**Comment by micheleg** [comment]
This should be fixed now in master [b702f055cff2cea18c89fea0592db479277a779c].
